### PR TITLE
Rewrite http webhook api

### LINKED ISSUE #466: [rewrite] Implement HTTP Webhook API
Ref: https://discord.com/developers/docs/resources/webhook

### PR DESCRIPTION
**Pull request by abitofevrything**
# Description

Implement the HTTP Webhook resource API

Prerequisite: #469

## Connected issues & other potential problems

Closes #466

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my changes haven't lowered code coverage

### CODE REVIEW

File: lib/src/builders/sentinels.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+const sentinelImageBuilder = _SentinelImageBuilder();
+
+class _SentinelImageBuilder implements ImageBuilder {
+  const _SentinelImageBuilder();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
```
Comment (l7ssha) [inline lib/src/builders/sentinels.dart:53]: why it needs to be implemented?

Comment (abitofevrything) [reply lib/src/builders/sentinels.dart:53]: Sentinels exist to be markers for a given type and not carry any implementation.

We could implement every member to throw `UnimplementedError` or we can use a Dart feature that if a class implements `noSuchMethod`, then it does not need to implement any other members.

This leads to shorter code with ultimately the same result, as no method should ever be called on a sentinel instance.

File: lib/src/http/managers/webhook_manager.dart [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+      ..messages(id: messageId.toString());
+    final request = BasicRequest(
+      route,
+      method: 'DELETE',
+      queryParameters: {if (threadId != null) 'thread_id': threadId.toString()},
+      authenticated: false,
+    );
+
+    await client.httpHandler.executeSafe(request);
+    // TODO: Can we delete the message from the cache?
```
Comment (l7ssha) [inline lib/src/http/managers/webhook_manager.dart:291]: I think message should be deleted here, yes

**Review by l7ssha — APPROVED** [review]

File: lib/src/http/managers/webhook_manager.dart (hunk 2 above)
Comment (abitofevrything) [reply lib/src/http/managers/webhook_manager.dart:291]: Yeah - the issue is we don't have the channel id to get the correct cache to remove it from

Comment (l7ssha) [reply lib/src/http/managers/webhook_manager.dart:291]: Then let's leave it for later. Doesn't matter that much I think

Comment (abitofevrything) [reply lib/src/http/managers/webhook_manager.dart:291]: We'll probably also get a gatetway event for the message deletion, so we can handle it there.

**Pull request merged**